Dockerfile

```diff
@@ -0,0 +1,11 @@
+# Stage 1: Build Go application
+FROM golang:1.20-alpine3.17 AS builder
+
+WORKDIR /app
+COPY . .
```

Comment:
I think it would be nice to do layer caching here. Each command in the Dockerfile becomes a new layer on the image. So if we change the image, the dependencies would be reinstalled. You could reestructure the Dockerfile so that we copy the dependencies _first_ and then copy everything else, like so:

```dockerfile
...
COPY go.mod go.sum ./
RUN go mod download

COPY . .
...
```

That way, we would recreate the dependencies only if we changed `go.sum`/`go.mod` and builds would be faster too. Here are some docs if you want to check this out: https://docs.docker.com/build/cache/, https://docs.docker.com/get-started/09_image_best/#layer-caching